### ISSUE TITLE
Opensource license?

### ISSUE THREAD

**Issue by firefu-mc** [Amulet-Team/Amulet-Core#155]
Hi,
I can't see any indication of which license this code is published under - has that been decided?
Many thanks

**Comment by gentlegiantJGC** [comment]
The licence is still TBD but we are working on guidelines for what it will be

**Comment by Redhacker1** [comment]
Sorry to necro, I am going to preface this with I am not a fan of the GPL, that being said I am not going to bash it in this comment. It looks like they were kind of heading towards MIT from the looks of it, so I am not sure GPL is really where they want to go. Honestly, GPL licensing aside from my personal issues with it. It does not seem like an inherently bad choice. Ultimately though it is up to them to decide and I am sure they will pick a license that goes with the philosophies of the developers and the project. @MestreLion

**Comment by ShayBox** [comment]
License needs to be decided soon, people are submitting PRs to an unlicensed project which can undermine some licenses, it also leaves packages, build scripts, binaries, forks, etc in an unknown grey area and discourages contribution.

**Comment by Redhacker1** [comment]
lemme preface this with IANAL: It does not leave it in a grey area IIRC. If it ain't licensed in many places including the US you just cannot use it or download it etc. All things not outlined by the license/developers are prohibited by default. If you could defend that position considering the circumstances and where it is uploaded and how in court, I do not know. However, as it stands downloading, compiling, modifying, etc when it comes to amulet is technically piracy and illegal.

**Comment by MestreLion** [comment]
> Ultimately though it is up to them to decide and I am sure they will pick a license that goes with the philosophies of the developers and the project.

Indeed, it's their choice. GPL was just a _suggestion_ I made, but in the end _any_ license, be it GPL, MIT, BSD, Apache, etc would greatly encourage contributors (myself included) and benefit the Minecraft community in general.

> It looks like they were kind of heading towards MIT from the looks of it, so I am not sure GPL is really where they want to go.

I saw no indication from Amulet's team to support either statement. From @gentlegiantJGC 's comment it seemed they were open to ideas and suggestions, and not leaning towards any particular license. Not sure why GPLv3 wouldn't be a suitable candidate, as would any free and opensource one.

**Comment by MestreLion** [comment]
> it also leaves packages, build scripts, binaries, forks, etc in an unknown grey area

> lemme preface this with IANAL: It does not leave it in a grey area IIRC. If it ain't licensed in many places including the US you just cannot use it or download it etc. All things not outlined by the license/developers are prohibited by default.

IANAL either, but AFAIK it is a _known_ area: in most (if not all) countries that have laws regarding intellectual property, a software without any license is covered by regular copyright law, where the default rule is _**"all rights reserved"**_. Even if the source code is published here, any _use_ of such code, including even _running_ it, is a copyright violation enforceable in court.

**Comment by Redhacker1** [comment]
> > Ultimately though it is up to them to decide and I am sure they will pick a license that goes with the philosophies of the developers and the project.
> 
> Indeed, it's their choice. GPL was just a _suggestion_ I made, but in the end _any_ license, be it GPL, MIT, BSD, Apache, etc would greatly encourage contributors (myself included) and benefit the Minecraft community in general.
> 
> > It looks like they were kind of heading towards MIT from the looks of it, so I am not sure GPL is really where they want to go.
> 
> I saw no indication from Amulet's team to support either statement. From @gentlegiantJGC 's comment, it seemed they were open to ideas and suggestions, and not leaning towards any particular license. Not sure why GPLv3 wouldn't be a suitable candidate, as would any free and open-source one.

My apologies I was thinking of a different project it seems. Yeah, It is not a bad license. It is a perfectly reasonable candidate.

**Comment by bekaertruben** [comment]
Hey, this seems like a pretty important issue you're just ignoring.
And it gets worse the longer you wait. If I recall correctly, you technically need to ask permission to change licence from every single contributor you've ever had, as the code is currently their property. Legally speaking, the current state is an utter mess. (I am no legal expert, though, so this may be entirely wrong)

Any of the popular FOSS licences are fine, if you want it to be copyleft go with the GPL (honestly, not my personal preference), otherwise something like MIT or BSD.

**Comment by Redhacker1** [comment]
It very much is, I've seen many projects, including dead ones that I would be very much open to contributing/adopting into my own stuff, lack of a license and communication killed off all security and incentive for doing so.

**Comment by gentlegiantJGC** [comment]
#192
